File: lambda/disconnect.py
import os
import boto3

ddb = boto3.resource("dynamodb")
table = ddb.Table(os.environ["TABLE_NAME"])

def handler(event,context):
    print(event)
    try:
        table.delete_item(
            Item={
                "ConnectionId": event["requestContext"]["connectionId"]
            }
        )
    except:
        pass
    return { "statusCode": 200 }

File: lambda/disconnect.js


const AWS = require('aws-sdk');

const ddb = new AWS.DynamoDB.DocumentClient({ apiVersion: '2012-08-10', region: process.env.AWS_REGION });

exports.handler = async (event) => {
  const tableName = process.env.TABLE_NAME;

  if (!tableName) {
    throw new Error('tableName not specified in process.env.TABLE_NAME');
  }

  const putParams = {
    TableName: tableName,
    Item: {
      ConnectionId: event.requestContext.connectionId,
    },
  };

  try {
    await ddb.delete(putParams).promise();
  } catch (err) {
    return { statusCode: 500, body: 'Failed to disconnect: ' + JSON.stringify(err) };
  }

  return { statusCode: 200, body: 'Disconnected.' };
};